### DISCUSSION TITLE
auto disable on password input

### DISCUSSION THREAD

**Discussion by mozartilize** [tadeokondrak/anthywl#22] [Q&A]
Hey, do you have any plan to support auto disable on password input? Or any advice how to implement it?

**Comment by tadeokondrak** [comment]
I tested out some applications and I don't seem to really get content type or content purpose hints. (using WAYLAND_DEBUG=1) There might be something wrong with my setup. If it was working though I would check `anthywl_seat::content_type_purpose == ZWP_TEXT_INPUT_V3_CONTENT_PURPOSE_PASSWORD`. Hope that helps

**Reply by mozartilize** [reply]
I'm thinking that it is application responsibility to set the `content_type_purpose`. I've asked on foot terminal repo[0], if I can get any help from the author.

Thank you!

[0] https://codeberg.org/dnkl/foot/issues/1330

**Comment by mozartilize** [comment]
foot wouldn't do it. So, I'm looking around. Sudo has plugin apis and maybe I will try if it could have a hook on before password prompt to to trigger disable ime or not.

Would you like to share your use cases on sudo? Do you manually disable ime on typing sudo password?

**Reply by tadeokondrak** [reply]
With a Japanese IME, ideally the IME would be disabled automatically whenever a terminal is focused and only enabled with a keybind (as Unix commands are in Latin characters). I type English much more often than I do Japanese so my current setup is just to toggle anthywl on temporarily whenever I need to type anything in Japanese.

I believe that on macOS you see a special key icon as part of the terminal when sudo is asking for a password. I wonder if they do anything special there with a terminal extension, and if it does anything for IMEs for other languages.

**Reply by mozartilize** [reply]
![image](https://user-images.githubusercontent.com/26346270/234333703-68af383d-ee38-4400-95f8-e47e5bf17a6c.png)

gnome-console also has that but I guess they just check the command if it contains `sudo` or not
